staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -246,6 +251,15 @@ func compileRule(rule apiextensions.ValidationRule, envSet *environment.EnvSet,
 		compilationResult.MessageExpression = msgProg
 		compilationResult.MessageExpressionMaxCost = costEst.Max
 	}
+	if rule.FieldPath != "" {
+		fldPath := field.NewPath("")
+		validFieldPath, err := ValidFieldPath(rule.FieldPath, fldPath, s)
```

Comment:
I thought the normalized field path would be a string (like "foo.bar.baz") we would then use like `fldPath.Child(NormalizedRuleFieldPath)`

That would mean:
1. we could pass in nil here instead of `field.NewPath("")` (so we don't get a leading `.`)
2. assign `compilationResult.NormalizedRuleFieldPath = validFieldPath.String()` (so we pay the string joining cost once)
3. use like `fldPath.Child(normalizedRuleFieldPath)` (so we don't need the SetRoot addition)

Reply:
My only qualm with using `.Child()` this way is that `Child()` is not smart enough t construct the underlying field path right. I.e. `fldPath.Child("a.b") != fieldPath.Child("a").Child("b")`  because the first one constructs a single node (with a property named "a.b") and the second one constructs two nodes.

Reply:
Maybe for now we just do `fldPath.Child(normalizedRuleFieldPath)` and then I can circle back after freeze and clean it up.

Reply:
if they have properties with dots in the name, they would need to express that as `foo['a.b']`, right? which we'd append as `preceeding.field.path.foo['a.b']`

Reply:
> the first one constructs a single node (with a property named "a.b") and the second one constructs two nodes

but they stringify the same, and get stringified before they get returned in the API response, which is what we really care about

Reply:
for what it's worth, we have existing use that calls Child a single time to construct multiple field levels:

```
git grep -E 'Child\("[^"]*\.' | grep -v test
pkg/apis/apps/validation/validation.go:		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(replicaStartOrdinal), fldPath.Child("ordinals.start"))...)
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:						allErrs = append(allErrs, field.Invalid(fldPath.Child("openAPIV3Schema.type"), schema.Type, fmt.Sprintf(`only "object" is allowed as the type at the root of the schema if the status subresource is enabled`)))
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:			allErrs = append(allErrs, field.Forbidden(fldPath.Child("openAPIV3Schema.nullable"), fmt.Sprintf(`nullable cannot be true at the root`)))
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:			allErrs = append(allErrs, field.Required(fldPath.Child("scale.specReplicasPath"), ""))
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:			if errs := validateSimpleJSONPath(subresources.Scale.SpecReplicasPath, fldPath.Child("scale.specReplicasPath")); len(errs) > 0 {
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:				allErrs = append(allErrs, field.Invalid(fldPath.Child("scale.specReplicasPath"), subresources.Scale.SpecReplicasPath, "should be a json path under .spec"))
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:			allErrs = append(allErrs, field.Required(fldPath.Child("scale.statusReplicasPath"), ""))
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:			if errs := validateSimpleJSONPath(subresources.Scale.StatusReplicasPath, fldPath.Child("scale.statusReplicasPath")); len(errs) > 0 {
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:				allErrs = append(allErrs, field.Invalid(fldPath.Child("scale.statusReplicasPath"), subresources.Scale.StatusReplicasPath, "should be a json path under .status"))
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:			if errs := validateSimpleJSONPath(*subresources.Scale.LabelSelectorPath, fldPath.Child("scale.labelSelectorPath")); len(errs) > 0 {
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go:				allErrs = append(allErrs, field.Invalid(fldPath.Child("scale.labelSelectorPath"), subresources.Scale.LabelSelectorPath, "should be a json path under either .spec or .status"))
```